omero/developers/Tables.rst

```diff
@@ -301,6 +301,99 @@ Examples
+
+  from getpass import getpass
+  from random import random
+  from omero.gateway import BlitzGateway
+  from omero.grid import LongColumn, StringColumn, DoubleColumn, ImageColumn
+  from omero.model import FileAnnotationI, OriginalFileI, ImageAnnotationLinkI, DatasetAnnotationLinkI, ImageI, DatasetI
+
+
+  def create_table(connection, dataset_id):
+      dataset = connection.getObject('Dataset', dataset_id)
+      images = [connection.getObject('Image', child.getId()) for child in dataset.listChildren()]
+
+      # Create some data
+      data = []
+      for image in images:
+          data.append(
+              {
+                  'image': image.getId(),
+                  'image_name': image.getName(),
+                  'random_number': random(),
+              }
+          )
+
+      # Determine column definitions
+      n = len(data)
+      cols = []
+      for key, value in data[0].items():
+          if key == 'image':
+              cols.append(ImageColumn(key, '', [0] * n))
+          elif isinstance(value, int):
+              cols.append(LongColumn(key, 'Integer value column', [0] * n))
+          elif isinstance(value, float):
+              cols.append(DoubleColumn(key, '', [0.0] * n))
+          elif isinstance(value, str):
+              cols.append(StringColumn(key, 'String col to store text', 64, ['-'] * n))
+
+      # initialize table
+      resources = connection.c.sf.sharedResources()
+      repository_id = resources.repositories().descriptions[0].getId().getValue()
+      table = resources.newTable(repository_id, "test:%i" % int(random()*1e6))
+      table.initialize(cols)
+
+      # Populate the columns
+      for col in cols:
+          for row_ind, row in enumerate(data):
+              col.values[row_ind] = row[col.name]
+
+      # Upload the data
+      table.addData(cols)
+      table_file_id = table.getOriginalFile().id.val
+      table.close()
+
+      # link the table file to the image
+      annotation = FileAnnotationI()
+      annotation.setFile(OriginalFileI(table_file_id, False))
+      annotation = connection.getUpdateService().saveAndReturnObject(annotation)
+      link = DatasetAnnotationLinkI()
+      link.setParent(DatasetI(dataset.getId(), False))
+      link.setChild(FileAnnotationI(annotation.getId().getValue(), False))
+      link = connection.getUpdateService().saveAndReturnObject(link)
+      print(link)
+
+
+  if __name__ == "__main__":
+      user_name = ""   # Username for the connection to the OMERO.server
+      host_name = ""   # DNS of the OMERO.server
+      group_id = ""    # OMERO group ID
+      dataset_id = ""  # Dataset with some test images in it
+
+      password = getpass(prompt='OMERO account password: ')
+      gateway = BlitzGateway(user_name, password, host=host_name, secure=True)
+      try:
+          gateway.connect()
+          gateway.keepAlive()
+          gateway.SERVICE_OPTS.setOmeroGroup(group_id)
+          gateway.setGroupForSession(group_id)
```

Comment:
Instead of requiring `group_id` as an input, you can first lookup the Dataset since by default the gateway will do a cross-group query (or you can enforce that with `gateway.SERVICE_OPTS.setOmeroGroup(-1)`).
Then you can get the group from the Dataset and use this, which you will need for saving table etc.
```
    group_id = dataset.getDetails().group.id.val
    conn.SERVICE_OPTS.setOmeroGroup(group_id)
```